internal/pkg/cachecli/cache_list.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cachecli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+func findSize(size int64) string {
+	var sizeF float64
+	if size <= 10000 {
+		sizeF = float64(size) / 1000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Kb"}, "")
+	} else if size <= 1000000000 {
+		sizeF = float64(size) / 1000000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Mb"}, "")
+	} else if size >= 1000000000 {
+		sizeF = float64(size) / 1000000000
+		return strings.Join([]string{fmt.Sprintf("%.2f", sizeF), " Gb"}, "")
+	}
+	return "ERROR: failed to detect file size."
```

Comment:
We shouldn't ever return error information encoded in a string return value. Once you change the function signature here according to the above comment, you can just do this:

`return "", fmt.Errorf("failed to detect file size")` And then we can determine how to handle a file whose size we cannot compute from wherever we call this function